pkg/proxy/handlers/spacelister.go

```diff
@@ -34,25 +35,59 @@ func NewSpaceLister(app application.Application) *SpaceLister {
 }
 
 func (s *SpaceLister) HandleSpaceListRequest(ctx echo.Context) error {
-	userID, _ := ctx.Get(context.SubKey).(string)
-	username, _ := ctx.Get(context.UsernameKey).(string)
+
+	workspaces, err := s.ListUserWorkspaces(ctx)
+	if err != nil {
+		return errorResponse(ctx, apierrors.NewInternalError(err))
+	}
+
 	workspaceName := ctx.Param("workspace")
 	doGetWorkspace := len(workspaceName) > 0
 
+	if doGetWorkspace { // specific workspace requested
+		if len(workspaces) != 1 {
+			// not found
+			r := schema.GroupResource{Group: "toolchain.dev.openshift.com", Resource: "workspaces"}
+			return errorResponse(ctx, apierrors.NewNotFound(r, workspaceName))
+		}
+		return getWorkspaceResponse(ctx, workspaces[0])
+	}
+
+	return listWorkspaceResponse(ctx, workspaces)
+}
+
+func (s *SpaceLister) ListUserWorkspaces(ctx echo.Context) ([]toolchainv1alpha1.Workspace, error) {
+	userID, _ := ctx.Get(context.SubKey).(string)
+	username, _ := ctx.Get(context.UsernameKey).(string)
```

Comment:
are those inputs from the "user"? is there any validation required or already in place for those?